src/org/apache/commons/httpclient/HttpMethodBase.java

```diff
@@ -1368,6 +1372,84 @@ protected void addCookieRequestHeader(HttpState state, HttpConnection conn)
     }
 
     /**
+     * Put all the cookies in a single header line. 
+     * 
+     * Merge the cookies already present in the request 
+     * with the cookies coming from the state. 
+     * 
+     * @param state the {@link HttpState state} information associated with this method
+     * @param conn the {@link HttpConnection connection} used to execute
+     *        this HTTP method
+     * @param cookies associated with the {@link HttpState state}
+     * 
+     */
+	private void putAllCookiesInASingleHeader(HttpState state, HttpConnection conn,
```

Comment:
The parameters could/should be `String host, CookieSpec matcher`, respectively, instead. They were already extracted in the previous method, so it just needs to pass what's really required.